2_base_and_sensor.ipynb

```diff
@@ -436,135 +812,259 @@
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Sensor values are stored using Python \"data structure\".\n",
+    "\n",
+    "The contents of the data structure can be accessed via \"[variable name].[structure element name]\"."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "例えば、`angle_min`, `angle_max`には、レーザスキャナのスキャン範囲（-120度から120度）がラジアンで格納されています。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "For example, `angle_min` and` angle_max` contain the scan range of the laser scanner (-120 to 120 degrees) in radians:"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# scan_data.angle_maxの値をラジアンからdegに変換します\n",
+    "# scan_data.angle_maxの値をラジアンからdegに変換します Convert value of scan_data.angle_max from radians to deg\n",
     "scan_data.angle_max / math.pi * 180  # math.pi = π"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "レーザスキャナのデータ本体は、`ranges`という名前の配列に格納されており、配列の長さは721のようです。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Actual laser scanner data is stored in the array named `ranges`, and the length of the array seems to be 721:"
```

Comment:
"The actual laser scanner data is stored in the array named `ranges`, and the length of the array seems to be 721:"